tools/shared.py

```diff
@@ -802,5 +829,11 @@ def init():
 EM_NM = bat_suffix(path_from_root('emnm'))
 FILE_PACKAGER = bat_suffix(path_from_root('tools/file_packager'))
 WASM_SOURCEMAP = bat_suffix(path_from_root('tools/wasm-sourcemap'))
+# Windows .dll suffix is not included in this list, since those are never
+# linked to directly on the command line.
+DYNAMICLIB_ENDINGS = ['.dylib', '.so']
+STATICLIB_ENDINGS = ['.a']
+
+run_via_emxx = False
```

Comment:
This is a little odd in `shared` I think? It only matters for emcc?

Reply:
It is used by both emcc.py and link.py.. it also need to be settabe from em++.py.

We could change thins into a function that checks sys.argv[0] I guess?